WcaOnRails/db/migrate/20231103103433_add_user_groups.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+class AddUserGroups < ActiveRecord::Migration[7.0]
+  def change
+    delegate_regions_user_group = UserGroup.create!(name: "Delegate Regions", group_type: :delegate_regions, is_active: true, is_hidden: false)
+    senior_delegates = User.where(delegate_status: "senior_delegate")
+    senior_delegates.each do |delegate|
```

Comment:
```suggestion
    User.find_each(delegate_status: "senior_delegate") do |delegate|
```

Reply:
Context: `find_each` is a Rails ActiveRecord method that automatically iterates over rows in a DB, but limits the batch size to 1000.

Your previous approach loads _all_ rows into memory, and _then_ starts looping over them 1 by 1. In case of Senior Delegates, this is not a problem because we only have a very small number of Senior Delegates in our DB, so it won't cause memory problems. But I think in general it's important to know about `find_each` because it helps you when there's a lot of entries.

Reply:
Okay got it, thanks for this. Changed.